plugin/pkg/shoot/validator/admission.go

```diff
@@ -391,6 +391,11 @@ func (c *validationContext) validateScheduling(ctx context.Context, a admission.
 				return admission.NewForbidden(a, fmt.Errorf("cannot schedule shoot '%s' on seed '%s' that already has the maximum number of shoots scheduled on it (%d)", c.shoot.Name, c.seed.Name, allocatableShoots.Value()))
 			}
 		}
+
+		err := c.validateSeedSelectionForHAShoot(a)
```

Comment:
This validation should be performed if `shootIsBeingScheduled` or `shootIsBeingRescheduled` are true, currently it is only performed when `shootIsBeingScheduled` is true.

@shafeeqes do you think it is ok to keep it only here. It doesn't make too much sense to also add `validateSeedSelectionForHAShoot()` to the `binding` admission plugin

Reply:
I did not check the PR thoroughly but if it is validating seed, I think it should be there in case of rescheduling as well, ie; in `binding` admission plugin.

Reply:
done

Reply:
this change was moved to #6530 and was merged recently. The check is changed.